### PR TITLE
626 pr feedback

### PR DESCRIPTION
**Pull request by MikeHathaway**
# Description of change
## High level
* Updates bucket bankruptcy check inequality as suggested by Dmitry.
  * Discovered in pr feedback here: https://github.com/ajna-finance/contracts/pull/625

### CODE REVIEW

File: src/PositionManager.sol [hunk 1]
```diff
@@ -242,9 +242,15 @@ contract PositionManager is ERC721, PermitERC721, IPositionManager, Multicall, R
             uint256 bucketDeposit,
         ) = IPool(params_.pool).bucketInfo(params_.fromIndex);
 
-        // check that bucket hasn't gone bankrupt since memorialization
-        if (positions[params_.tokenId][params_.fromIndex].depositTime < bankruptcyTime) {
-            revert BucketBankrupt();
+        // check for previous deposits
+        if (positions[params_.tokenId][params_.fromIndex].depositTime != 0) {
```
Comment (MikeHathaway) [inline src/PositionManager.sol]: A whole lot of SLOAD operations here reading from Position, but running into stack too deep issues when attempting to use variables.

**Review by EdNoepel — APPROVED** [review]
LGTM

**Review by ith-harvey — APPROVED** [review]
LGTM

File: src/PositionManager.sol [hunk 2]
```diff
@@ -242,9 +242,17 @@ contract PositionManager is ERC721, PermitERC721, IPositionManager, Multicall, R
...
+        // check for previous deposits
+        {
+            uint256 depositTime = positions[params_.tokenId][params_.fromIndex].depositTime;
+            if (depositTime != 0) {
+                // check that bucket hasn't gone bankrupt since memorialization
+                if (depositTime <= bankruptcyTime) {
+                    revert BucketBankrupt();
+                }
+            } else {
+                revert RemovePositionFailed();
```
Comment (grandizzy) [inline src/PositionManager.sol]: why is this needed here?

File: src/PositionManager.sol [hunk 3]
```diff
@@ -280,6 +288,8 @@ contract PositionManager is ERC721, PermitERC721, IPositionManager, Multicall, R
         // update position LPs state
         positions[params_.tokenId][params_.fromIndex].lps -= lpbAmountFrom;
         positions[params_.tokenId][params_.toIndex].lps   += lpbAmountTo;
+        // update position deposit time to the from bucket deposit time
+        positions[params_.tokenId][params_.toIndex].depositTime = positions[params_.tokenId][params_.fromIndex].depositTime;
```
Comment (grandizzy) [inline src/PositionManager.sol]: can add a struct to hold local vars and not need this

**Review by dmitriia — COMMENTED** [review]
Looks ok
